### DISCUSSION TITLE
How to make my website available off-line?

### DISCUSSION THREAD

**Discussion by vaughanwatson** [datenstrom/community#186] [Ask a question] [answered]
I found part of the answer here 
[http://developers.datenstrom.se/help/web-server-configuration#static-website](http://developers.datenstrom.se/help/web-server-configuration#static-website)

and here 

[http://www.html5rocks.com/en/tutorials/appcache/beginner/](http://www.html5rocks.com/en/tutorials/appcache/beginner/)

Basically what I need to do is
1. Convert pages to a static website
2. Create an app.cache file

My users should then be able to continue reading a document when they are on a train and internet connections drop as they go into a tunnel.

**Comment by markseuffert** [comment]
I am afraid to say that I don't know how to make a website available offline with AppCache. As far as I understand from this [AppCache article](http://alistapart.com/article/application-cache-is-a-douchebag) it's not a solution for every website, because you have to jump trough hoops. You could build a static website and a [cache manifest](http://marcotroisi.com/make-your-website-available-offline-with-html5/), which includes all necessary media/system files. Whenever your website is updated you make a new release with a new manifest.

@dominikwilkowski do you have some tips?

**Comment by vaughanwatson** [comment]
Thanks Mark,

As  I don't have access to a php command line (everything is tied down
tight), your suggestion of getting Yellow to generate a static website,
then I create a cache manifest, which a developer can then upload to the
server and a business manager can access these policies and procedures html
pages anywhere seems better.

I'll have a look at this over the weekend and see if I can get it working.

Thanks for your feedback, much appreciated.

Vaughan

**Comment by dominikwilkowski** [comment, marked as answer]
There are many other ways you can go "offline first" but to know the best is to know your circumstances.
(Refer to a talk I held the other day for the W3C specs: https://dominikwilkowski.github.io/talks/a11y/#/3/1)

**Comment by dieli** [comment]
@vaughanwatson Do you have php command line or not? I needed to convert my yellow site to a static site once without php command line, and I just created a php script which I then navigated to in my browser:

```
<?php
  require_once("system/core/core.php");
  $yellow = new Yellow();
  $yellow->plugins->load();
  $statusCode = $yellow->command("commandline", "build");
?>
```

**Comment by vaughanwatson** [comment]
markseu, dominikwilkowski and dieli ,

thank you guys for helping me push through to a possible solution to this problem. I really do appreciate learning from your comments.

I'll have a crack at it over the next few days and let you know how I get on.
